javascript-style-guide.md

```diff
@@ -143,6 +179,131 @@ algorithms that might grow the call stack boundlessly.
 Aggregate operations provide forms for iterating over collections that are more
 precise and oftentimes more useful than their procedural alternatives.
 
+## Object-orientation
+
+You can leverage object-oriented design patterns such as encapsulation,
+composition, and inheritance in JavaScript. Create functions whose instance
+variables represent private and protected members. From these functions, return
+objects with methods to access and possibly manipulate the instance
+variables. Create functions with extended interfaces that mutate instances of
+their parents to effect inheritance.
+
+Example:
+
+```js
+var makeMammal = function (data) {
+    var self = {};
+    self.getName = function () {
+        return data.name;
+    };
+    self.says = function () {
+        return data.saying === undefined ? '' : data.saying;
+    };
+    return self;
+};
+
+var myMammal = makeMammal({name: 'Herb'});
+
+var makeCat = function (data) {
+    var self;
+    data.saying = data.saying === undefined ? 'meow' : data.saying;
+    self = makeMammal(data);
+    self.purr = function (n) {
+        var message = '',
+            index;
+        for (index = 0; index < n; index += 1) {
```

Comment:
I agree, I considered using `times` instead. For simplicity and copy-paste-ability, it seemed to better to use this.